I need (and have implemented) support for Request.Files

mike-schenk opened the pull request:
I added one test, but I didn't cover all the code I added with tests.

I also simplified the querystring parsing code.

mike-schenk committed:
Added support for Request.Files

mike-schenk committed:
Added support for Request.Path and an implementation of Request.FilePath

mike-schenk committed:
Added support for Response.StatusCode, Response.StatusDescription and Response.SubStatusCode

mikevalenty commented on src/Web/FakeHttpFileCollection.cs:
```diff
@@ -0,0 +1,219 @@
+			// Unfortunately, HttpFileCollectionBase overrides lots of features of NameObjectCollectionBase by throwing NotImplementedExceptions.
+			// It also fails to use a case-insensitive key-comparer the way the real HttpFileCollection does.
+			// So we'll wrap our own NameObjectCollectionBase implementation instead.
+			mWrappedCollection = new ConcreteNameObjectCollection();
```
You captured an important requirement (case-insensitive) in the form of a comment, this would be an excellent behavior of FakeHttpFileCollection to unit test.

mikevalenty commented on src/Web/FakeHttpFileCollection.cs:
```diff
@@ -0,0 +1,219 @@
+{
+	public class FakeHttpFileCollection : HttpFileCollectionBase
+	{
+		private readonly ConcreteNameObjectCollection mWrappedCollection;
```
Hungarian notation "mWrappedCollection" in not used anywhere else in the rest of the code base.

mikevalenty commented on src/Web/FakeHttpPostedFile.cs:
```diff
@@ -0,0 +1,49 @@
+namespace FakeN.Web
+{
+	public class FakeHttpPostedFile : HttpPostedFileBase
+	{
```
Hungarian notation ("m" prefix) is not consistent with the rest of the project

mikevalenty commented on src/Web/FakeHttpPostedFile.cs:
```diff
@@ -0,0 +1,49 @@
+		}
+
+		public override void SaveAs(string filename)
+		{
```
Users of this library will likely not want to write to the actual file system during a unit test, so it might be better to just set a property like SavedFileName for use in assertions and possibly add a method like `OnSaveAs(Action<String> action)` that would allow the user supply a lambda for whatever mock behavior they want for the `SaveAs` method.

mikevalenty commented on src/Web/FakeHttpRequest.cs:
```diff
@@ -145,16 +146,34 @@ namespace FakeN.Web
 
 		static NameValueCollection ParseQueryString(string url)
 		{
```
Much simpler!

mikevalenty commented:
Thanks for the patch! I made a few comments and I look forward to merging your contribution.

mike-schenk committed:
Added support for Request.InputStream.
Added unit test for case-insensitivity on FakeHttpFileCollection.
Made the FakeHttpPostedFile.SaveAs method unit-test friendly.
Fixed naming convention problem in FakeHttpPostedFile.cs

mike-schenk committed:
Fixed naming convention problem in FakeHttpFileCollection.cs

mike-schenk commented:
In these last two commits, I believe I have resolved the issues you pointed out.
I also have added one more property to the Request object, InputStream

mike-schenk committed:
Clarified a code comment, renamed the inner class and renamed a field to make it more consistent with the rest of the codebase.
